dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/jobService/JobFacade.kt

```diff
@@ -17,19 +17,17 @@ package com.spectralogic.dsbrowser.gui.services.jobService
 import com.spectralogic.ds3client.Ds3Client
 import io.reactivex.Completable
 import io.reactivex.Observable
-import io.reactivex.Observer
 import javafx.beans.property.LongProperty
 import javafx.beans.property.SimpleBooleanProperty
-import java.time.Instant
 import java.util.*
-import java.util.concurrent.atomic.AtomicLong
+import kotlin.reflect.KFunction0
 
 interface JobFacade {
     public fun titleObservable(): Observable<String>
     public fun jobSizeObservable(): Observable<Number>
     public fun messageObservable(): Observable<String>
     public fun visabilityObservable(): Observable<Boolean>
-    public fun finishedCompletable(): Completable
+    public fun finishedCompletable(cancelled: KFunction0<Boolean>): Completable
```

Comment:
Why are you using a `KFunction0<Boolean>` instead of a `Consumer<Boolean>`

Reply:
That's what Intelij typed it as, I'll change it.

Reply:
I need the result this returns, Consumer doesn't give me back the value.

Reply:
Then use a Supplier